shared/src/ledger/pos/vp.rs

```diff
@@ -415,6 +416,23 @@ where
             .unwrap();
         decode(value).unwrap()
     }
+
+    fn read_validator_eth_cold_key(
+        &self,
+        key: &Self::Address,
+    ) -> Option<types::ValidatorEthKey<Self::PublicKey>> {
+        let value =
+            self.ctx.read_pre(&validator_eth_cold_key_key(key)).unwrap();
+        value.map(|value| decode(value).unwrap())
+    }
+
+    fn read_validator_eth_hot_key(
+        &self,
+        key: &Self::Address,
+    ) -> Option<types::ValidatorEthKey<Self::PublicKey>> {
+        let value = self.ctx.read_pre(&validator_eth_hot_key_key(key)).unwrap();
+        value.map(|value| decode(value).unwrap())
+    }
```

Comment:
it can happen if we are a fullnode, for instance

Reply:
Even for a full node, we are reading the validator's key from blockchain state (`self.ctx.read_pre`) when executing this VP, so these two funcs should work the same for full nodes and validators? I guess returning an `Option` here follows the pattern of other methods here for the time being

Reply:
at type level `key: &Self::Address` can be any address, so this would be `None` if the given address is not a validator address. If we know we're calling it with validator addresses only, we can expect that those are always `Some`